src/main/java/io/strimzi/kafka/metrics/YammerMetricsCollector.java

```diff
@@ -0,0 +1,148 @@
+
+    private final List<MetricsRegistry> registries;
+    private final PrometheusMetricsReporterConfig config;
+
+    public YammerMetricsCollector(PrometheusMetricsReporterConfig config) {
+        this.config = config;
+        this.registries = Arrays.asList(KafkaYammerMetrics.defaultRegistry(), Metrics.defaultRegistry());
+    }
+
+    @Override
+    public List<MetricFamilySamples> collect() {
+        List<MetricFamilySamples> samples = new ArrayList<>();
+
+        for (MetricsRegistry registry : registries) {
+            for (Map.Entry<MetricName, Metric> entry : registry.allMetrics().entrySet()) {
+                MetricName metricName = entry.getKey();
+                Metric metric = entry.getValue();
+                LOG.trace("Collecting Yammer metric {}", metricName);
+
+                String name = metricName(metricName);
+                // TODO Filtering should take labels into account
+                if (!config.isAllowed(name)) {
+                    LOG.info("Yammer metric {} is not allowed", name);
+                    continue;
+                }
+                LOG.info("Yammer metric {} is allowed", name);
+                Map<String, String> labels = labelsFromScope(metricName.getScope());
+                LOG.info("labels " + labels);
+
+                MetricFamilySamples sample = null;
+                if (metric instanceof Counter) {
+                    sample = convert(name, (Counter) metric, labels);
+                } else if (metric instanceof Gauge) {
+                    sample = convert(name, (Gauge<?>) metric, labels);
+                } else if (metric instanceof Histogram) {
+                    sample = convert(name, (Histogram) metric, labels);
+                } else if (metric instanceof Meter) {
+                    sample = convert(name, (Meter) metric, labels);
+                } else if (metric instanceof Timer) {
+                    sample = convert(name, (Timer) metric, labels);
+                } else {
+                    LOG.error("The metric " + metric.getClass().getName() + " has an unexpected type.");
+                }
+                if (sample != null) {
+                    samples.add(sample);
+                }
+            }
+        }
+        return samples;
+    }
+
+    static String metricName(MetricName metricName) {
+        String metricNameStr = Collector.sanitizeMetricName(
+                "kafka_server_" +
+                metricName.getGroup() + '_' +
+                metricName.getType() + '_' +
+                metricName.getName()).toLowerCase();
+        LOG.info("metricName group {}, type {}, name {} converted into {}", metricName.getGroup(), metricName.getType(), metricName.getName(), metricNameStr);
+        return metricNameStr;
+    }
+
+    static Map<String, String> labelsFromScope(String scope) {
+        if (scope != null) {
+            String[] parts = scope.split("\\.");
```

Comment:
We're paying the cost of compiling this regex with every call. It's probably worth pulling out and reusing a `Pattern`

Reply:
Yes, that's a potential performance improvement we should try